src/overlays/fbdemos/ovl_fbdemo_triforce/z_fbdemo_triforce.c

```diff
@@ -15,35 +16,135 @@ void TransitionTriforce_SetType(void* thisx, s32 type);
-    TransitionTriforce_Start,
-    TransitionTriforce_SetType,
-    TransitionTriforce_SetColor,
-    NULL,
+    TransitionTriforce_Init,   TransitionTriforce_Destroy, TransitionTriforce_Update,   TransitionTriforce_Draw,
+    TransitionTriforce_Start,  TransitionTriforce_SetType, TransitionTriforce_SetColor, NULL,
     TransitionTriforce_IsDone,
 };
 
-#endif
+void TransitionTriforce_Start(void* thisx) {
+    TransitionTriforce* this = (TransitionTriforce*)thisx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_fbdemo_triforce/TransitionTriforce_Start.s")
+    switch (this->state) {
+        case 1:
+        case 2:
+            this->transPos = 1.0f;
+            return;
+    }
+    this->transPos = 0.03f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_fbdemo_triforce/TransitionTriforce_Init.s")
+void* TransitionTriforce_Init(void* thisx) {
+    TransitionTriforce* this = (TransitionTriforce*)thisx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_fbdemo_triforce/TransitionTriforce_Destroy.s")
+    bzero(this, sizeof(*this));
+    guOrtho(&this->projection, -160.0f, 160.0f, -120.0f, 120.0f, -1000.0f, 1000.0f, 1.0f);
+    this->transPos = 1.0f;
+    this->state = 2;
+    this->step = 0.015f;
+    this->fadeDirection = 1;
+    return this;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_fbdemo_triforce/TransitionTriforce_Update.s")
+void TransitionTriforce_Destroy(void* thisx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_fbdemo_triforce/TransitionTriforce_SetColor.s")
+void TransitionTriforce_Update(void* thisx, s32 updateRate) {
+    TransitionTriforce* this = (TransitionTriforce*)thisx;
+    f32 temp_f0;
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_fbdemo_triforce/TransitionTriforce_SetType.s")
+    for (i = updateRate; i > 0; i--) {
+        if (this->state == 1) {
+            this->transPos = CLAMP_MIN(this->transPos * (1.0f - this->step), 0.03f);
+        } else if (this->state == 2) {
+            this->transPos = CLAMP_MIN(this->transPos - this->step, 0.03f);
+        } else if (this->state == 3) {
+            this->transPos = CLAMP_MAX(this->transPos / (1.0f - this->step), 1.0f);
+        } else if (this->state == 4) {
+            this->transPos = CLAMP_MAX(this->transPos + this->step, 1.0f);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_fbdemo_triforce/func_80AC5280.s")
+void TransitionTriforce_SetColor(void* thisx, u32 color) {
+    TransitionTriforce* this = (TransitionTriforce*)thisx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_fbdemo_triforce/TransitionTriforce_Draw.s")
+    this->color.rgba = color;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_fbdemo_triforce/TransitionTriforce_IsDone.s")
+void TransitionTriforce_SetType(void* thisx, s32 type) {
+    TransitionTriforce* this = (TransitionTriforce*)thisx;
+
+    this->fadeDirection = type;
+}
+
+void func_80AC5280(void* thisx, s32 state) {
```

Comment:
```suggestion
void TransitionTriforce_SetState(void* thisx, s32 state) {
```
Identical to OoT, so might as well.